storage/include/storage/db_types.h

```diff
@@ -50,8 +50,10 @@ enum class EDBKeyType : std::uint8_t {
 // containing actual application data.
 enum class EKeySubtype : std::uint8_t {
   Internal,
-  Stale,
+  ProvableStale,
+  NonProvableStale,
```

Comment:
In order to preserve compatibility with an existing DB, I think we need to add new types at the end of the enum. In particular, adding `NonProvableStale` before `Leaf` will change the value for `Leaf`.

Reply:
Another thing came to mind - would be nice if we can test that somehow in a unit test. Will be relevant for migrations, as a whole.

Reply:
Added a `static_assert` in the header file.

Reply:
I was thinking that maybe we can be explicit about the values themselves in the enum declaration. And maybe we can put a comment to say that we can't change the existing ones.

I also had a look at the file and the comment at the bottom about key ordering is no longer correct (the one with the ASCII art). I think it would be better to remove it and just comment on the enum itself that we cannot change the existing values.